File: backend/internal/models/custom_team_template.go
package models

import "time"

type CustomTeamTemplate struct {
	ID                   int       `db:"id,primarykey,autoincrement" json:"id"`
	UserID               int       `db:"user_id" json:"user_id"`
	Name                 string    `db:"name" json:"name"`
	Intent               string    `db:"intent" json:"intent"`
	RequestedMemberCount *int      `db:"requested_member_count" json:"requested_member_count,omitempty"`
	ResolvedMemberCount  int       `db:"resolved_member_count" json:"resolved_member_count"`
	SpecJSON             string    `db:"spec_json" json:"-"`
	Revision             int       `db:"revision" json:"revision"`
	LastTraceID          *string   `db:"last_trace_id" json:"-"`
	CreatedAt            time.Time `db:"created_at" json:"created_at"`
	UpdatedAt            time.Time `db:"updated_at" json:"updated_at"`
}

func (CustomTeamTemplate) TableName() string {
	return "custom_team_templates"
}

File: frontend/src/types/customTeamTemplate.ts
export interface CustomTeamMemberSpec {
  memberId: string;
  displayName: string;
  role: string;
  isLeader: boolean;
  summary: string;
  mission: string;
  responsibilities: string[];
  boundaries: string[];
  expectedInputs: string[];
  deliverables: string[];
  acceptanceCriteria: string[];
  collaborationNotes: string[];
  capabilityTags: string[];
}

export interface CustomTeamTemplateSpec {
  schemaVersion: number;
  name: string;
  summary: string;
  resolvedMemberCount: number;
  members: CustomTeamMemberSpec[];
}

export interface CustomTeamTemplate {
  id: number;
  name: string;
  intent: string;
  requested_member_count?: number;
  resolved_member_count: number;
  revision: number;
  spec: CustomTeamTemplateSpec;
  created_at: string;
  updated_at: string;
}
